tests/workflow/test_resolve_interface.py

```diff
@@ -0,0 +1,153 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for the `qml.workflow.resolution._resolve_interface` helper function"""
+
+import pytest
+
+import pennylane as qml
+from pennylane.tape import QuantumScript
+from pennylane.workflow import _resolve_interface
+
+
+def test_auto_with_numpy():
+    """Test that 'auto' interface resolves to numpy correctly."""
+    tapes = [
+        QuantumScript([qml.RX(0.5, wires=0)], [qml.expval(qml.PauliZ(0))]),
+    ]
+    resolved_interface = _resolve_interface("auto", tapes)
+    assert resolved_interface == "numpy"
+
+
+@pytest.mark.tf
+def test_auto_with_tf():
+    """Test that 'auto' interface resolves to 'tf' correctly."""
+    try:
+        # pylint: disable=import-outside-toplevel
+        import tensorflow as tf
+    except ImportError:
+        pytest.skip("TensorFlow is not installed.")
+    tapes = [
+        QuantumScript([qml.RX(tf.Variable(0.5), wires=0)], [qml.expval(qml.PauliZ(0))]),
+    ]
+    resolved_interface = _resolve_interface("auto", tapes)
+    assert resolved_interface == "tf"
+
+
+def test_auto_with_unsupported_interface():
+    """Test that 'auto' interface resolves to None correctly."""
+    # pylint: disable=import-outside-toplevel
+    import networkx as nx
+
+    # pylint: disable=too-few-public-methods
+    class DummyCustomGraphOp(qml.operation.Operation):
+        """Dummy custom operation for testing purposes."""
+
+        def __init__(self, graph: nx.Graph):
+            super().__init__(graph, wires=graph.nodes)
+
+        def decomposition(self) -> list:
+            return []
+
+    graph = nx.complete_graph(3)
+    tape = qml.tape.QuantumScript([DummyCustomGraphOp(graph)], [qml.expval(qml.PauliZ(0))])
+
+    assert _resolve_interface("auto", [tape]) is None
+
+
+@pytest.mark.jax
+def test_auto_with_jax():
+    """Test that 'auto' interface resolves to 'jax' correctly.."""
+    try:
+        # pylint: disable=import-outside-toplevel
+        import jax.numpy as jnp
+    except ImportError:
+        pytest.skip("JAX not installed.")
+
+    tapes = [
+        QuantumScript([qml.RX(jnp.array(0.5), wires=0)], [qml.expval(qml.PauliZ(0))]),
+    ]
+    resolved_interface = _resolve_interface("auto", tapes)
+    assert resolved_interface == "jax"
+
+
+@pytest.mark.tf
+def test_tf_autograph():
+    """Test that 'tf' interface resolves to 'tf-autograph' in graph mode."""
+    try:
+        # pylint: disable=import-outside-toplevel
+        import tensorflow as tf
+    except ImportError:
+        pytest.skip("TensorFlow is not installed.")
+
+    # pylint: disable=not-context-manager
+    with tf.Graph().as_default():
+        tapes = [
+            QuantumScript([qml.RX(tf.constant(0.5), wires=0)], [qml.expval(qml.PauliZ(0))]),
+        ]
+        resolved_interface = _resolve_interface("tf", tapes)
+
+    assert resolved_interface == "tf-autograph"
```

Comment:
@albi3ro, do you know of a better way to mimic a `tf-autograph` tape?

Reply:
nope. autograph is just painful all around.

Reply:
😿 ... resolving.